index.js

```diff
@@ -1,3 +1,228 @@
-/** @module audio-speaker */
-'use strict';
-module.exports = require('./direct');
+/** @module  audio-speaker/index */
+
+'use strict'
+
+var os = require('os')
+var objectAssign = require('object-assign')
+var binding = require('audio-mpg123')
+var pcm = require('pcm-util')
+var isAudioBuffer = require('is-audio-buffer')
+var audioSink = require('audio-sink/direct')
+var debug = require('debug')('speaker')
+
+var endianess = 'function' == os.endianess ? os.endianess() : 'LE'
+
+module.exports = Speaker
+
+/**
+ * The Speaker function initializes a new audio handler,
+ * then returns the write method to output audio to.
+ *
+ * @param {Object} opts options for the speaker
+ * @return {Function} write write audio from a buffer or audiobuffer
+ * @module Speaker
+ * @api public
+ */
+function Speaker (opts) {
+  debug('Speaker()')
+  var options = {}
+
+  options = objectAssign({
+    channels: 1,
+    float: false,
+    bitDepth: 16,
+    signed: true,
+    samplesPerFrame: 1024,
+    sampleRate: 44100,
+    endianess: endianess,
+    autoFlush: false
+  }, opts)
+
+  if (options.handler) {
+    throw new Error('_create() was called more than once. Only one handler should exist.')
+  }
+
+  options._closed = false
+  options._busy = false
+
+  var format = Speaker.getFormat(options)
+  if (format === null) {
+    throw new Error('Invalid format options.')
+  }
+
+  options.blockAlign = options.bitDepth / 8 * options.channels
+
+  options.chunkSize = options.blockAlign * options.samplesPerFrame
+
+  options.handler = binding.create((success) => {
+    if(!success) {
+      throw new Error('Failed to create the audio handler.')
+    } else {
+      debug('_create() audio handle successfully.')
+    }
+  })
+
+  if (options.handler !== null) {
+    debug('_start(%o)', Object.keys(options))
+    binding.open(options.handler, options.sampleRate, options.channels, format, function (success) {
+      if (!success) {
+        throw new Error('Could not start the audio output with these properties.')
+      } else {
+        debug('Created and started handler successfully.')
+      }
+    })
+  }
+
+  write.end = end
+  write.sink = sink
+  sink.end = end
+  return write
+
+  /**
+   * The write function takes a buffer or audiobuffer and
+   * writes it to the speaker output. If the chunks are too
+   * large it will break it up and put the remainding chunks
+   * into a queue.
+   * NOTE: You can only write new chunks once the callback is
+   * called with no errors.
+   *
+   * @param {AudioBuffer} chunk (or Buffer) containing the data to be output
+   * @param {Function} callback callback with error and chunk parameters
+   * @return void
+   * @api public
+   */
+  function write (chunk, callback) {
```

Comment:
Could `callback` be optional?

In quick scripts I don't really care to test when `write` is done.  End result is to output to speaker.  `foo()(speaker())`.

Gives me `TypeError` otherwise.